### PR TITLE
feat(ctx-menu): SW-side chrome.contextMenus integration (part of #72)

### PR DESCRIPTION
**Pull request by chriscantu**
## Summary

Implements the **SW-side surface** of issue #72 context-menu integration per the Accepted spec at `docs/superpowers/specs/2026-05-25-context-menu-integration.md` (2026-05-26).

**Mini-modal scope-swap ACs deferred** — overlay infrastructure (issue #19) not yet shipped; CS is a 19-line stub. AC #10 (scoped header), #11 (← Full article swap), #15 (empty-selection live-region), #16 (focus on mount) wait for #19. Tracking those in a follow-up sub-issue.

### Wave dispatch (ruflo swarm hierarchical, queen=extension-architect, swarmId swarm-1779809081123-red2yn)

- **Wave 1A** (`coder`): `SettingsSchemaV4` + V3→V4 migration + storage type swap. 94 tests.
- **Wave 1B** (`coder`): `OverridesSchema` + `validateOverrides` + `pickValidOverrides`. 33 tests.
- **Wave 1C** (`chrome-extension-engineer`): `ContextMenuActivationIntent.menuItemId` + `overrides`; `ActivateReaderMessage.overrides?`; payload normalization. 4 new dispatch tests, collision test threaded.
- **Wave 2A** (`chrome-extension-engineer`): `factory.ts` / `install.ts` / `listener.ts` / `register.ts`; manifest `contextMenus` permission. 30 new tests.
- **Final reviewer** (`reviewer`): APPROVE, 2 non-blocking nits fixed (V4 import swap + title copy alignment with spec).

### In-scope AC pins

| AC | Pinned by |
|----|-----------|
| #1 contextMenus.create + http(s) patterns | `install.test.ts:77-92` |
| #2 Eight v1 IDs in order | `factory.test.ts:20-32` |
| #3 lastUsed title | `factory.test.ts:34-44` |
| #4 Preset → overrides.wpm | `listener.test.ts:90-121`, `dispatch.test.ts` |
| #5 Custom → openOptionsPage | `listener.test.ts:139-148` |
| #6 Toggle → saveSettings, no dispatch | `listener.test.ts:150-162` + `install.test.ts:117-133` |
| #7 startFromWord1 toggle | `listener.test.ts:164-176` |
| #8 Backwards-compat (popup/command omit overrides) | `dispatch.test.ts:243-279` |
| #9 V3→V4 migration | `migrations.test.ts:335-349` |
| #12 Restricted-URL guard (documentUrlPatterns) | factory + adapter |
| #13 onStartup loadSettings ordering | `install.test.ts:152-178` |
| #14 #34 hotkey collision | `collision.test.ts` (in-flight Map preserved) |
| #17 Override bounds gate (wpm:999999 dropped) | `validate-overrides.test.ts` |
| #18 Frame provenance | `listener.test.ts:178-197` |
| #19 Top-level sync listener registration | `register.test.ts:48-82` |

### Deferred ACs (blocked by #19 overlay)

- AC #10 — scoped mini-modal `SELECTION · N words · ~M sec` header
- AC #11 — `← Full article` in-overlay swap
- AC #15 — empty-selection live-region status
- AC #16 — `role=dialog` / `aria-modal` / focus-on-play-pause mount contract

These will land in a follow-up PR once #19 ships the overlay infrastructure. Listener already sends `ctx-frame-blocked` toast messages; CS-side rendering arrives with overlay.

## Test plan

- [x] `npm run lint` — exits 0, zero warnings
- [x] `npm run format:check` — "All matched files use Prettier code style!"
- [x] `npx tsc --noEmit` — exits 0
- [x] `npm test -- --run` — 27 files, 478 tests, all passing (verified 2026-05-26)
- [x] `npm run build` — `dist/manifest.json` contains `"contextMenus"`
- [ ] Load unpacked extension in Chrome (`chrome://extensions`), confirm no install errors and `contextMenus` listed in permissions — MANUAL, pending human verification
- [ ] Right-click on a selection on an https:// page → "SpeedReader" parent visible with submenu (last-used / 300 / 500 / Custom / separator / Show context line / Start from word 1) — MANUAL, pending human verification
- [ ] Right-click on `chrome://settings` → SpeedReader entry NOT visible (documentUrlPatterns guard) — MANUAL, pending human verification
- [ ] Click `Custom…` → Options page opens; no overlay activates — MANUAL, pending human verification
- [ ] Toggle `Show context line` → `chrome.storage.sync` shows `contextLine: true`; menu reopens with checked state; no overlay activates — MANUAL, pending human verification
- [ ] Click `300 wpm` → console logs `ctx-frame-blocked` is NOT emitted (top-frame); SW dispatches `activate-reader` with `overrides.wpm: 300` (verify via SW devtools); `lastUsedWpm` persists to 300 — MANUAL, pending human verification
- [ ] Right-click inside an iframe → submenu appears, click any item → no activation, `ctx-frame-blocked` message sent — MANUAL, pending human verification

## Constraints honored

- `src/core/**` boundary: zero `chrome.*` imports (factory.ts in `src/chrome/` also pure-TS)
- MV3 listener-registration discipline: all 4 listeners registered top-level synchronously in `register.ts`
- Surgical scope: all 26 changed files trace to spec sections
- Backwards-compat: popup-source and command-source activations produce byte-identical payloads to pre-PR

Closes part of #72 (SW surface). Follow-up issue: #131 (mini-modal scope-swap, blocked by #19).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Verification status (2026-05-26)

**Automated gates** (5/5 ✅): lint, format:check, tsc, vitest (478/478), build w/ `contextMenus` permission. Run from `/Users/cantu/repos/speedreader-chrome` on commit d6102b8.

**Manual gates** (7/7 ⏳): require operator-driven Chrome browser interaction (load unpacked, right-click, devtools inspection). Agent cannot perform — flagged per `rules/pr-validation.md` "items unverifiable on host MUST be flagged explicitly". Reviewer/author runs these before merge.

### CODE REVIEW

File: src/core/messaging/validate.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  return result.success ? result.data : null;
+}
+
+/**
+ * Per-field bounds gate. Walks an already-shape-valid `Overrides` and drops
+ * any field that fails its per-field schema (out-of-bound `wpm`, non-integer
+ * `wpm`, non-`multipleOf(10)` `wpm`, non-boolean toggles). Returns the
+ * remaining valid subset (possibly empty). Never throws.
+ */
+export function pickValidOverrides(raw: Overrides): Overrides {
```
Comment (chriscantu) [inline src/core/messaging/validate.ts:65]: Why use raw type instead of a specific type. Seems like we are having to do a lot of type handling for a typed language.

File: src/core/settings/index.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 export {
+  SettingsSchemaV4,
   SettingsSchemaV3,
   SettingsSchemaV2,
   CURRENT_VERSION,
   VALID_ALIGNMENTS,
+  type SettingsV4,
```
Comment (chriscantu) [inline src/core/settings/index.ts:7]: Do we need to keep track of schema settings at this point since we have not shipped a customer version yet?

File: src/core/settings/migrations.ts [hunk 3]
```diff
@@ -46,7 +61,7 @@ const MIGRATIONS: Record<number, Migrator> = {
  * `'migrates v3 blob with missing field by filling defaults (explicit repair
  * behavior)'` in `__tests__/migrations.test.ts`.
  */
-export function migrate(rawValue: unknown): SettingsV3 {
+export function migrate(rawValue: unknown): SettingsV4 {
```
Comment (chriscantu) [inline src/core/settings/migrations.ts:74]: why use "raw" types when using a typed language?

**Comment by chriscantu** [comment]
Addressed PR review (5-reviewer pass: code-reviewer, pr-test-analyzer, silent-failure-hunter, type-design-analyzer, comment-analyzer) in commit d6102b8.

**Critical fixed:**
- Field-name mismatch `overrides.showContext` → `contextLine` (shallow-merge no-op bug)
- Silent void-fire-and-forget chains across register.ts ×4 + listener.ts dispatchActivation Result discards → all sites now log
- `chrome.contextMenus.create/update` unchecked → callback-based `runtime.lastError` checks added

**Important fixed:**
- Re-entrancy race in `ensureContextMenu` → in-flight Promise lock
- `migrate()` lastUsedWpm propagated bogus raw.wpm → `clampLastUsedWpm` helper
- Diff path silently dropped new IDs → warn log
- Separator/checkbox property guards in adapter

**Test coverage gaps closed:** +5 tests (478 total)
- register: subscribeSettings handler invocation verified
- migrations: lastUsedWpm clamp boundary cases (3)
- dispatch: AC #17 funnel-level forged-overrides forwarding

**Deferred to follow-ups:**
- Type-design refactor recommendations → #132
- frame-targeted `sendMessage` with `frameId` → blocked by #19 (#131)

Gates: lint clean, format clean, tsc clean, 478/478 tests pass, build emits `contextMenus` permission.
